### PR TITLE
Remove ReceiveAsync/CompleteAsync style public APIs for c2d messaging

### PR DESCRIPTION
**Pull request by timtay-microsoft**
In the interest of keeping these PRs small, removing the internal methods that follow this pattern will be tackled separately later

Updated the relevant samples and migration doc with this context as well.

### CODE REVIEW

File: SDK v2 migration guide.md [hunk 1]
```diff
@@ -52,6 +50,7 @@ to migrate to version 2.x when they have the chance. For more details on LTS rel
   - It had very limited support across the device options and some APIs behaved differently.
 - Some options that were previously set in the `DeviceClient` constructor are now in the optional `IotHubClientOptions` parameter.
 - This method has been split into the three individual steps that this method used to take. See [this file upload sample](./iothub/device/samples/getting%20started/FileUploadSample/) for an example of how to do file upload using these discrete steps.
+- Cloud to device messages can only be received by calling ```SetReceiveMessageHandlerAsync``` and providing a callback. Users can no longer poll for messages with ```ReceiveAsync```.
```
Comment (drwill-ms) [inline SDK v2 migration guide.md:53]: Triple back ticks are only needed for code blocks. Code snippets can be single back ticks.

```suggestion
- Cloud to device messages can only be received by calling `SetReceiveMessageHandlerAsync` and providing a callback. Users can no longer poll for messages with `ReceiveAsync`.
```

File: SDK v2 migration guide.md [hunk 2]
```diff
@@ -52,6 +50,7 @@ to migrate to version 2.x when they have the chance. For more details on LTS rel
   - It had very limited support across the device options and some APIs behaved differently.
 - Some options that were previously set in the `DeviceClient` constructor are now in the optional `IotHubClientOptions` parameter.
 - This method has been split into the three individual steps that this method used to take. See [this file upload sample](./iothub/device/samples/getting%20started/FileUploadSample/) for an example of how to do file upload using these discrete steps.
+- Cloud to device messages can only be received by calling `SetReceiveMessageHandlerAsync` and providing a callback. Users can no longer poll for messages with `ReceiveAsync`.
```
Comment (drwill-ms) [inline SDK v2 migration guide.md:53]:
```suggestion
- Cloud-to-device messages can be received by calling `SetReceiveMessageHandlerAsync` and providing a callback. Users no longer need to poll for messages with `ReceiveAsync`.
```

File: e2e/test/iothub/service/MessageFeedbackReceiverE2ETest.cs [hunk 3]
```diff
@@ -62,17 +63,32 @@ public async Task MessageFeedbackReceiver_Operation(IotHubTransportProtocol prot
...
                     new IotHubClientOptions(new IotHubClientAmqpSettings()));
                 await deviceClient.OpenAsync().ConfigureAwait(false);
-                Client.Message receivedMessage = await deviceClient.ReceiveMessageAsync().ConfigureAwait(false);
-                await deviceClient.CompleteMessageAsync(receivedMessage.LockToken).ConfigureAwait(false);
 
-                Task result = await Task
+                var c2dMessageReceived = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                Func<Client.Message, object, Task<MessageAcknowledgement>> OnC2DMessageReceived = (message, context) =>
+                {
+                    c2dMessageReceived.SetResult(true);
```
Comment (drwill-ms) [inline e2e/test/iothub/service/MessageFeedbackReceiverE2ETest.cs:70]:
```suggestion
                    c2dMessageReceived.TrySetResult(true);
```

File: e2e/test/iothub/messaging/MessageFeedbackE2ETests.cs [hunk 4]
```diff
@@ -1,82 +0,0 @@
...
-using Microsoft.Azure.Devices.Client;
-using Microsoft.Azure.Devices.E2ETests.Helpers;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
-
-namespace Microsoft.Azure.Devices.E2ETests.Messaging
-{
-    [TestClass]
-    [TestCategory("E2E")]
-    [TestCategory("IoTHub")]
-    public class MessageFeedbackE2ETests : E2EMsTestBase
```
Comment (timtay-microsoft) [inline e2e/test/iothub/messaging/MessageFeedbackE2ETests.cs:19]: This test class doesn't cover anything interesting anymore now that the receiveAsync call is gone and the callbacks are executed in order. We have a separate test class for the actual message feedback receiving part of this test

File: iothub/device/src/IotHubDeviceClient.cs [hunk 5]
```diff
@@ -230,12 +205,37 @@ private async Task OnDeviceMessageReceivedAsync(Message message)
...
+
+                            case MessageAcknowledgement.Abandon:
+                                await InnerHandler.AbandonMessageAsync(message.LockToken, CancellationToken.None).ConfigureAwait(false);
+                                break;
+
+                            case MessageAcknowledgement.Reject:
+                                await InnerHandler.RejectMessageAsync(message.LockToken, CancellationToken.None).ConfigureAwait(false);
+                                break;
+
+                            default:
```
Comment (drwill-ms) [inline iothub/device/src/IotHubDeviceClient.cs:231]: Don't need the default case.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**